src/pybind/mgr/dashboard/frontend/src/app/shared/forms/cd-validators.spec.ts

```diff
@@ -902,5 +902,52 @@ describe('CdValidators', () => {
         testValidator('testName', true);
       }));
     });
+
+    describe('endpointValidator', () => {
+      it('should return null for a valid URL with port', () => {
+        const control = new UntypedFormControl('https://example.com:8080');
+        expect(CdValidators.endpointValidator(control)).toBeNull();
+      });
+
+      it('should return null for multiple valid URLs with ports', () => {
+        const control = new UntypedFormControl('https://example.com:8080,http://localhost:3000');
+        expect(CdValidators.endpointValidator(control)).toBeNull();
+      });
+
+      it('should return an error object for a URL without a port', () => {
+        const control = new UntypedFormControl('https://example.com');
+        expect(CdValidators.endpointValidator(control)).toEqual({ invalidEndpoint: true });
+      });
+
+      it('should return an error object for multiple invalid URLs', () => {
+        const control = new UntypedFormControl('https://example.com,http://invalid-url');
+        expect(CdValidators.endpointValidator(control)).toEqual({ invalidEndpoint: true });
+      });
+
+      it('should return an error object for a non-URL string', () => {
+        const control = new UntypedFormControl('randomstring');
+        expect(CdValidators.endpointValidator(control)).toEqual({ invalidEndpoint: true });
+      });
+
+      it('should return null for a valid IP address with port', () => {
+        const control = new UntypedFormControl('https://192.168.1.1:9090');
+        expect(CdValidators.endpointValidator(control)).toBeNull();
+      });
+
+      it('should return an error object for an IP address without a port', () => {
+        const control = new UntypedFormControl('https://192.168.1.1');
```

Comment:
this is valid btw. any reason not to be valid? does the back-end throw any error if its an implicit port?